app_dart/lib/src/service/build_status_provider.dart

```diff
@@ -81,14 +59,58 @@ class BuildStatusService {
               /// the build status to fail.
               tasksInProgress[task.name] = false;
             } else if (_isFailed(task) || _isRerunning(task)) {
-              return BuildStatus.failed;
+              failedTasks?.add(task.name);
```

Comment:
This stops the algorithm from failing fast. Do we know what the performance implications would be?